addons/l10n_be_pos_sale/models/pos_session.py

```diff
@@ -0,0 +1,12 @@
+from odoo import models
+
+
+class PosSession(models.Model):
+    _inherit = 'pos.session'
+
+    def _pos_data_process(self, loaded_data):
+        res = super()._pos_data_process(loaded_data)
+        if self.company_id.country_code == 'BE':
+            intracom_fpos = self.env.ref("l10n_be.fiscal_position_template_3", False).with_company(self.company_id)
```

Comment:
I don't think you should reference the fiscal position template, but rather the fiscal position itself.

```suggestion
            intracom_fpos = self.env.ref(f"l10n_be.{self.env.company.id}_fiscal_position_template_3", False)
```

Also, if the xmlid isn't found, won't `env.ref` return `None` and you'll get an `AttributeError` when calling `with_company`?